src/pygama/dsp/processors/presum.py

```diff
@@ -3,15 +3,19 @@
 import numpy as np
 from numba import guvectorize
 
+from pygama.dsp.errors import DSPFatal
 from pygama.dsp.utils import numba_defaults_kwargs as nb_kwargs
 
 
 @guvectorize(
-    ["void(float32[:], float32[:])", "void(float64[:], float64[:])"],
-    "(n),(m)",
+    [
+        "void(float32[:], float32, float32[:], float32[:])",
+        "void(float64[:], float64, float64[:], float64[:])",
+    ],
+    "(n), (), (), (m)",
     **nb_kwargs,
 )
-def presum(w_in: np.ndarray, w_out: np.ndarray) -> None:
+def presum(w_in: np.ndarray, do_norm: int, ps_fact: int, w_out: np.ndarray) -> None:
```

Comment:
don't we also need a version of presum that also takes in dt and makes an appropriate output version (dt/presum)?

Reply:
wait -- nevermind -- I remember now that processing_chain handles it internally, right?